src/Sulu/Bundle/AdminBundle/Resources/js/components/CircleSelection/CircleSelectionRenderer.js

```diff
@@ -0,0 +1,196 @@
+// @flow
+import React from 'react';
+import type {Normalizer, SelectionData} from './types';
+import ModifiableCircle from './ModifiableCircle';
+import PositionNormalizer from './normalizers/PositionNormalizer';
+import RoundingNormalizer from './normalizers/RoundingNormalizer';
+import SizeNormalizer from './normalizers/SizeNormalizer';
+
+type Props = {
+    containerHeight: number,
+    containerWidth: number,
+    disabled: boolean,
+    filled: boolean,
+    label?: string,
+    maxRadius?: number,
+    minRadius?: number,
+    onChange: (value: ?SelectionData) => void,
+    percentageValues: boolean,
+    resizable: boolean,
+    round: boolean,
+    value: SelectionData | typeof undefined,
+};
+
+export default class CircleSelectionRenderer extends React.Component<Props> {
+    static defaultProps = {
+        disabled: false,
+        filled: false,
+        percentageValues: false,
+        resizable: true,
+        round: true,
+    };
+
+    handleChange = (value: ?SelectionData) => {
+        const {onChange, containerWidth, containerHeight, percentageValues} = this.props;
+
+        if (!percentageValues || !value) {
+            onChange(value);
+
+            return;
+        }
+
+        const {left, top, radius = 0} = value;
+
+        onChange({
+            left: left / containerWidth,
+            top: top / containerHeight,
+            radius: radius / containerWidth,
+        });
+    };
+
+    get value() {
+        const {value, containerWidth, containerHeight, percentageValues} = this.props;
+
+        if (!value) {
+            return this.getMaximumSelection();
+        }
+
+        if (!percentageValues) {
+            return value;
+        }
+
+        const {left, top, radius = 0} = value;
+
+        return {
+            left: left * containerWidth,
+            top: top * containerHeight,
+            radius: radius * containerWidth,
+        };
+    }
+
+    componentDidMount() {
+        this.setInitialValue();
+    }
+
+    componentDidUpdate() {
+        this.setInitialValue();
+    }
+
+    setInitialValue = () => {
```

Comment:
This ensures that if you pass `undefined` for `this.props.value`, the `CircleSelectionRenderer` immediately calls `this.props.handleChange` with the default value. This ensures that `this.props.value` is never null

Reply:
If I would just do `const {value = this.getDefaultValue()} = this.props;` the Circle would be shown in the admin in the center of the container, but on the website the value would be still null.

Reply:
Okay, does that actually happen? Or to rephrase the question, when is the value going to be `null`? Aren't you setting something right after adding a new hotspot? Or do I remember that wrong?

Reply:
No, when adding a new hotspot `undefined` is passed as value to the `CircleSelectionRenderer`, because only the `CircleSelectionRenderer` itself knows it's default value

Reply:
Ok, makes sense. But is that also necessary for `componentDidUpdate`? There this case should actually never happen, right?